src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -484,12 +484,16 @@ private static List<ChatMessageContentItem> GetContentParts(IList<AIContent> con
                     parts.Add(new ChatMessageTextContentItem(textContent.Text));
                     break;
 
-                case ImageContent imageContent when imageContent.Data is { IsEmpty: false } data:
-                    parts.Add(new ChatMessageImageContentItem(BinaryData.FromBytes(data), imageContent.MediaType));
-                    break;
+                case DataContent dataContent when dataContent.HasMediaTypePrefix("image/"):
+                    if (dataContent.ContainsData)
+                    {
+                        parts.Add(new ChatMessageImageContentItem(BinaryData.FromBytes(dataContent.Data!.Value), dataContent.MediaType));
```

Comment:
I'm surprised the `!` is needed here on `Data!`. The MemberNotNullWhen on Data is supposed to address this:
https://github.com/dotnet/extensions/blob/6e20a080cb52b7a842610714118b35fb0baaf3dc/src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/DataContent.cs#L183-L185
Are you actually getting a warning here without the exclamation? If yes, might be worth following up with Roslyn folks to understand why.

Reply:
Yeah, I was getting a warning. Looks like the issue was that the `[MemberNotNullWhen]` attribute should have been on the `ContainsData` property, not the `Data` property.

After moving the attribute, the warning goes away (and a few other existing locations were made simpler too).